src/libLookup/Lookup.cpp

```diff
@@ -1248,6 +1248,247 @@ bool Lookup::ProcessSetSeedPeersFromLookup(const vector<unsigned char>& message,
     return true;
 }
 
+bool Lookup::AddMicroBlockToStorage(const uint64_t& blocknum,
+                                    const MicroBlock& microblock)
+{
+    uint32_t id = microblock.GetHeader().GetShardID();
+
+    TxBlock txblk = m_mediator.m_txBlockChain.GetBlock(blocknum);
+    unsigned int i = 0;
+
+    if (txblk == TxBlock())
+    {
+        LOG_GENERAL(WARNING, "Failed to fetch microblock");
+        return false;
+    }
+    for (i = 0; i < txblk.GetShardIDs().size(); i++)
+    {
+
+        if (txblk.GetShardIDs()[i] == id)
+        {
+            break;
+        }
+    }
+    if (i == txblk.GetShardIDs().size())
+    {
+        LOG_GENERAL(WARNING, "Failed to find id " << id);
+        return false;
+    }
+
+    if ((txblk.GetMicroBlockHashes()[i].m_txRootHash
+         == microblock.GetHeader().GetTxRootHash())
+        && (txblk.GetMicroBlockHashes()[i].m_stateDeltaHash
+            == microblock.GetHeader().GetStateDeltaHash()))
+    {
+        vector<unsigned char> body;
+        microblock.Serialize(body, 0);
+        if (!BlockStorage::GetBlockStorage().PutMicroBlock(blocknum, id, body))
+        {
+            LOG_GENERAL(WARNING, "Failed to put microblock in body");
+            return false;
+        }
+    }
+    else
+    {
+        LOG_GENERAL(WARNING, "MicroBlock not validated");
+        return false;
+    }
+
+    return true;
+}
+
+bool Lookup::ProcessGetMicroBlockFromLookup(
+    const vector<unsigned char>& message, unsigned int offset, const Peer& from)
+{
+    LOG_MARKER();
+
+    if (!LOOKUP_NODE_MODE)
```

Comment:
Same question here for the check. Can we remove it if we know this function won't be called wrongly? Or can we do the check in the caller so that the function can be cleaner?